gymnasium/spaces/text.py

```diff
@@ -69,7 +70,9 @@ def __init__(
         super().__init__(dtype=str, seed=seed)
 
     def sample(
-        self, mask: Optional[Tuple[Optional[int], Optional[np.ndarray]]] = None
+        self,
+        mask: None
+        | (tuple[int | None, np.ndarray[Any, np.dtype[np.int8]] | None]) = None,
```

Comment:
We should be consistent with either `np.ndarray` or `NDArray` in typing. 

Are you sure about the generic arguments to `np.ndarray` here?

Reply:
Good spot, updating